diff -Naurp Modules/Setup.dist Modules/Setup.dist
--- Modules/Setup.dist	2013-09-05 07:20:40.993710533 -0500
+++ Modules/Setup.dist	2013-09-05 07:21:19.157709099 -0500
@@ -202,10 +202,10 @@ GLHACK=-Dclear=__GLclear
 
 # Socket module helper for SSL support; you must comment out the other
 # socket line above, and possibly edit the SSL variable:
-#SSL=/usr/local/ssl
-#_ssl _ssl.c \
-#	-DUSE_SSL -I$(SSL)/include -I$(SSL)/include/openssl \
-#	-L$(SSL)/lib -lssl -lcrypto
+SSL=/usr
+_ssl _ssl.c \
+	-DUSE_SSL -I$(SSL)/include -I$(SSL)/include/openssl \
+	-L$(SSL)/lib -lssl -lcrypto
 
 # The crypt module is now disabled by default because it breaks builds
 # on many systems (where -lcrypt is needed), e.g. Linux (I believe).
